### PR TITLE
Fix nested IdOffsetRange constructor

### PR DESCRIPTION
**Pull request by jishnub**
There was a bug introduced by #178 in defining `IdOffsetRange(::IdOffsetRange, offset)`, this PR fixes it.

On master
```julia
julia> b = 1:10
1:10

julia> bov = OffsetArray(view(b, 3:4), 3:4);

julia> c = @view b[bov]
2-element view(::UnitRange{Int64}, 3:4 with indices 3:4) with eltype Int64 with indices 3:4:
 3
 4

julia> d = OffsetArray(c, 1:2)
2-element OffsetArray(view(::UnitRange{Int64}, 3:4 with indices 3:4), 1:2) with eltype Int64 with indices 1:2:
 1
 2
```

After this PR:
```julia
julia> d = OffsetArray(c, 1:2)
2-element OffsetArray(view(::UnitRange{Int64}, 3:4 with indices 3:4), 1:2) with eltype Int64 with indices 1:2:
 3
 4
```

### CODE REVIEW

**Comment by jishnub** [comment]
Fixes #186 , this was a bug in `OffsetArrays` after all. Now 

```julia
julia> a = reshape(1:12, 3, 4)
3×4 reshape(::UnitRange{Int64}, 3, 4) with eltype Int64:
 1  4  7  10
 2  5  8  11
 3  6  9  12

julia> view(a, :, OffsetArrays.IdOffsetRange(3:4)) == view(a, :, 3:4)
true
```

File: src/axes.jl [hunk 1]
```diff
@@ -78,6 +78,9 @@ struct IdOffsetRange{T<:Integer,I<:AbstractUnitRange{T}} <: AbstractUnitRange{T}
     offset::T
 
     IdOffsetRange{T,I}(r::I, offset::T) where {T<:Integer,I<:AbstractUnitRange{T}} = new{T,I}(r, offset)
+    function IdOffsetRange{T,IdOffsetRange{T,I}}(r::IdOffsetRange{T,I}, offset::T) where {T<:Integer,I<:AbstractUnitRange{T}}
```
Comment (timholy) [inline src/axes.jl:81]: Do you need this? AFAICT it does the same thing as the default. If this is for ambiguity resolution, perhaps add a comment?

File: src/axes.jl [hunk 2]
```diff
@@ -96,13 +99,12 @@ IdOffsetRange(r::AbstractUnitRange{T}, offset::Integer = 0) where T<:Integer =
 IdOffsetRange{T,I}(r::IdOffsetRange{T,I}) where {T<:Integer,I<:AbstractUnitRange{T}} = r
 function IdOffsetRange{T,I}(r::IdOffsetRange, offset::Integer = 0) where {T<:Integer,I<:AbstractUnitRange{T}}
     rc, offset_rc = offset_coerce(I, r.parent)
-    return IdOffsetRange{T,I}(rc, r.offset + offset + offset_rc)
+    return IdOffsetRange{T,I}(rc, convert(T, r.offset + offset + offset_rc))
```
Comment (timholy) [inline src/axes.jl:102]:
```suggestion
    return IdOffsetRange{T,I}(rc, convert(T, r.offset + offset + offset_rc)::T)
```
Most of the time that's a no-op, but if anyone supplies a funky `Integer` and then writes a broken `convert` method that doesn't actually return a `T` (there is nothing preventing that from happening), your version would yield a StackOverflowError. This yields a TypeError, which is more graceful.

**Review by timholy — APPROVED** [review]

**Comment by timholy** [comment]
Very good!

Would this have been easier to detect if IdOffsetRange printed the indices as well as values? I understand that not everyone likes #179, but should we change the printing some other way so that it is more revealing?

**Comment by jishnub** [comment]
Yes indeed, halfway through debugging this I was wishing that the indices were printed along with the values. It'll be good if we can figure out a way to display them.

File: src/axes.jl (hunk 1 above)
Comment (jishnub) [reply src/axes.jl:81]: This is to avoid a stack overflow error in another method. I'll leave a comment

Comment (timholy) [reply src/axes.jl:86]: Makes sense. Perhaps the argument-type coercion doesn't need to specialize on `IdOffsetRange` for the range? If that's not an easy fix, then just adding a brief comment would be a nice addition.

Comment (jishnub) [reply src/axes.jl:86]: I think the idea behind that method was to remove the `IdOffsetRange` wrapper and coerce the parent range, as there is another method for generic `AbstractUnitRange` arguments. Not quite sure of the performance impacts of removing this, perhaps one may look into it in another PR. As of now I have added a comment in 7f4eecb

**Pull request merged**